packages/gatsby-core-utils/src/service-lock.ts

```diff
@@ -46,10 +49,10 @@ export const getService = (
   name: string
 ): Promise<string | null> => {
   const lockfileDir = getLockfileDir(programPath)
-  const lockfilePath = path.join(lockfileDir, `${name}.lock`)
+  const datafilePath = path.join(lockfileDir, `${name}.lock`, `data`)
```

Comment:
```Javascript
  const lockfileDir = path.join(getLockfileDir(programPath), `${name}.lock`)
  const datafilePath = path.join(lockfileDir, `data`)
```
i suggest to have the `lockfileDir` the same like in line 29, so that it not looks like a bug in line 52 because it is different compared to line 42